source/mir/bignum/fixed.d

```diff
@@ -388,6 +388,61 @@ struct UInt(size_t size)
         return overflow;
     }
 
+    /++
+    Performs `remainder = big /= rhs` operatrion,
+    Params:
+        rhs = unsigned value to divide by
+    Returns:
+        unsigned remainder value
+    +/
+    UInt!rhsSize opOpAssign(string op : "/", size_t rhsSize)(UInt!rhsSize rhs)
+        @safe pure nothrow @nogc scope
+    {
+        import mir.bignum.low_level_view : divm, BigUIntView;
+
+        uint[(size / (uint.sizeof * 8)) + 1] _div;
+        // shouldn't be unaligned here -- maybe a possibility though
+        _div[0 .. $ - 1] = (cast(uint[])data)[0 .. $];
+        data[0 .. $] = 0;
+
+        BigUIntView!uint dividend = BigUIntView!uint(_div);
+        BigUIntView!uint divisor = cast(BigUIntView!uint)rhs.view;
+        BigUIntView!uint quotient = cast(BigUIntView!uint)view;
+        divm(dividend, divisor, quotient);
+
+        UInt!rhsSize rem = 0;
+        BigUIntView!uint remainder = cast(BigUIntView!uint)rem.view;
+        remainder.coefficients[0 .. $] = (cast(uint[])_div)[0 .. remainder.coefficients.length];
+
+        return rem;
+    }
+
+    static if (size == 128)
+    ///
+    version(mir_bignum_test)
+    @safe pure @nogc
+    unittest
+    {
+        auto a = UInt!128.fromHexString("e3251bacb112c88b71ad3f85a970a314");
+        auto b = UInt!128.fromHexString("dfbbfae3cd0aff2714a1de7022b0029d");
+        auto rem = a /= b;
+        assert(a == UInt!128.fromHexString("1"));
+        assert(rem == UInt!128.fromHexString("36920c8e407c9645d0b611586c0a077"));
+    }
+
+        static if (size == 128)
+    ///
+    version(mir_bignum_test)
+    @safe pure @nogc
+    unittest
+    {
+        auto a = UInt!128.fromHexString("e3251bacb112c88b71ad3f85a970a314");
+        auto b = UInt!64.fromHexString("dfbbfae3cd0aff27");
+        auto rem = a /= b;
+        assert(a == UInt!128.fromHexString("103e70bb237983337"));
+        assert(rem == UInt!64.fromHexString("4932c5448b170cb3"));
+    }
```

Comment:
Also we need `%=`, `/`, and `%` operations. And standalone `divRem` method that performs both division and remainder for Uints. All `/=`, `%=`, `/`, and `%` can call it instead of the `divm`.

Reply:
OK. I got this implemented in UInt -- should it also be implemented in BigInt?

Reply:
No